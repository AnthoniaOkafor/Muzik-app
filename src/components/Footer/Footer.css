footer{
    background: #1ABC9C 0% 0% no-repeat padding-box;
    border-radius: 10px;
    opacity: 1;
    margin: 0% 5% 12% 5% ;
    /* change padding to affect how social icons look */
    padding: 3% 2% 5% 4%;
    color: white;
    font-size: 125%;

}li{
    list-style-type: none;
}
.footerLink {
    display: flex;
}
/* This section is for different plaform Icon */
.download{
    font-size: 18px;
    margin: -110px 0px 0px 64%;
    
}
.download > ul > li{
    display: inline;
}
.download > ul {
    margin-top:17px;
}
.download > .listOfLink > li > a > img{
    width: 30px;
    padding-left: 18px;
}
/* This section is for different social media Icon */
.footerLink > span > ul {
    background-color: white;
    position: absolute;
    padding: 10px 15px;
    top: 9%;
    left: 97.1%;
}
.footerLink > span > .listOfLink > li > a > img{
    width: 30px;
    padding: 18px 0px;
    display: block
}
.footerLink {
    position: relative;

}
/* This section is for List of Links e.g blog */
.linkGroup > ul > li > a{
    text-decoration: none;
    color: white;
    line-height: 36px;
    font-size: 18px;
}
/* inserts space before every list */
.linkGroup > .listOfLink{
    margin: 20px 0px 20px;
}
/* inserts space after every list */
.linkGroup {
    margin-right: 21%;
}



