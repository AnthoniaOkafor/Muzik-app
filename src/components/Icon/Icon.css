.socials{
    background-color: white;
    position: absolute;
    padding: 10px 15px;
    top: 9%;
    right: 5%;
}
.socials > a > img{
    width: 30px;
    padding: 18px 0px;
    display: block

}
.platform{
    margin-top:17px;
}
.platform > a > img{
    width:30px;
    padding-left: 18px;
}