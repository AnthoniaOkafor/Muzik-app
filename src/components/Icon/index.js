import React from "react";


const Icon = ({ iconLinks, type }) => {  
    
    return(
        <div className={`icon ${type}`}>
        {iconLinks.map(({imgPath, url, alt}) => <a key={imgPath} href={url}><img src={imgPath} alt={alt}/></a>)}
        </div>        
    )
}

export default Icon