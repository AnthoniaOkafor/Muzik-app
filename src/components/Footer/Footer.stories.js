import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs/register';
import Footer from '.';


storiesOf('Footer', module)
	.addDecorator(
		withInfo({
			text: "This is a footer component"
		})
	)
	.add('footer', () => {
        return <Footer />
    })