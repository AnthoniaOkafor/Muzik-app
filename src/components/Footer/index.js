import React from "react"
import Icon from "../Icon/";
import ListOfLink from "../ListOfLink";

import Header from "../Header";
import Text from "../Text"
import "./Footer.css"

const android = "./assets/android.svg"
const apple = "./assets/apple.svg"
export const facebook = "./assets/facebook.svg"
const linkedin = "./assets/linkedin.svg"
const youtube = "./assets/youtube.svg"

export const platform = [{
        item: <Icon src={apple} alt={'apple icon'}/>,
        link: "/lfdjd"
    },{
        item: <Icon src={android} alt={'android icon'}/>,
        link: "#"
    }]
export let socials = [{
        item: <Icon src={facebook} alt={'facebook icon'}/>,
        link: 'https://www.facebook.com'
    },{
        item: <Icon src={linkedin} alt={'linkedin icon'}/>,
        link: 'https://www.linkedin.com'
    },{
        item: <Icon src={youtube} alt={'youtube icon'}/>,
        link: 'https://www.youtube.com'
    }]
export let link =[{
        Categories: ["All events", "Weddings", "Birthdays", "Karoake", "Clubs", "Get togethers"],
        Resources: ["Media resources", "Terms & conditions", "Privacy policy"],
        QuickLinks: ["Blogs", "Company", "Pricing"],
    }]

const Footer = () => { 

    let group = Object.entries(link[0]).map(([key, value]) => {
        return (
            <div key={key} className="linkGroup">
                <Header heading={key} size={3} color={"white"}/>
                {<ListOfLink item={value}/>}
            </div>
        )
    })
    
    return (
        <footer>
            <div className="footerLink">
             {group}
             <span>
             <ListOfLink item={socials}/>
             </span>
             
            </div>
            <div className="download">
                <Text text="Download App" size="extrasmall"/>
                <ListOfLink item={platform}/>
            </div>
        </footer>
    )

}

export default Footer