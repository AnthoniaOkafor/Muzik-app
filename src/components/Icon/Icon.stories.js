import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text } from "@storybook/addon-knobs";
import { facebook } from "../Footer";
import Icon from ".";

storiesOf("Icon", module)
  .addDecorator(
    withInfo({
      text: "This is a component that renders Icon"
    })
  )
  .addDecorator(withKnobs())
  .add("Icon", () => {
    return <Icon src={facebook} alt="facebook" />;
  })
  .add("your icon", () => {
    const src = text("path", "./link/to/icon");
    const alt = text("alt", "insert alt message");

    return <Icon src={src} alt={alt} />;
  });
