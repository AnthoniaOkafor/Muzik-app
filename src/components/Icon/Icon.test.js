import React from "react";
import { shallow } from "enzyme";
import { facebook } from "../Footer";

import Icon from ".";

const src = facebook;
const alt = "facebook";

const setUp = () => {
  return shallow(<Icon src={src} alt={alt} />);
};

describe("Icon Component", () => {
  let icon;
  beforeEach(() => {
    icon = setUp();
  });

  it("should match Icon snap shot", () => {
    expect(icon).toMatchSnapshot();
  });
  it("should render icon image correctly", () => {
    const image = icon.find("img");

    expect(image.length).toBe(1);
  });
});

// check property data type
