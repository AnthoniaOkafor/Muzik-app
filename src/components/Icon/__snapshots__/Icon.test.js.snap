// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should match Icon snap shot 1`] = `ShallowWrapper {}`;
